addons/binding/org.openhab.binding.gpio/src/main/java/org/openhab/binding/gpio/handler/GPIODigitalInputHandler.java

```diff
@@ -0,0 +1,125 @@
+        }
+    }
+
+    @Override
+    public void initialize() {
+        jPigpio = ((PigpioBridgeHandler) getBridge().getHandler()).getJPiGpio();
+        try {
+            gpio = new GPIO(jPigpio, Integer.parseInt(getConfig().get(GPIO_ID).toString()), 1);
+            jPigpio.gpioSetAlertFunc(gpio.getPin(), new Alert() {
+
+                @Override
+                public void alert(int gpio, int level, long tick) {
+                    lastChanged = new Date();
+                    new Thread(new Runnable() {
+
+                        @Override
+                        public void run() {
+
+                            try {
+                                Date thisChange = new Date();
+                                OnOffType state = getValue();
+                                Thread.sleep(((BigDecimal) getConfig().get(PULLUP_TIME)).longValue());
+
+                                if ((boolean) getConfig().get(PULLUP_STRICT_MODE)) {
+                                    // Check if value changed over time
+                                    if (!thisChange.before(lastChanged)) {
+                                        updateState(getThing().getChannel(THING_TYPE_DIGITAL_INPUT_CHANNEL).getUID(),
+                                                getValue());
+                                    }
+                                } else {
+                                    // Check if value is exact after time
+                                    if (state.equals(getValue())) {
+                                        updateState(getThing().getChannel(THING_TYPE_DIGITAL_INPUT_CHANNEL).getUID(),
+                                                getValue());
+                                    }
+                                }
+                            } catch (InterruptedException e) {
+                                logger.error("Unknown exception", e);
+                            } catch (PigpioException e) {
+                                logger.error("Unknown jpigpio exception", e);
+                            }
+                        }
+                    }).start();
+                }
+            });
+            logger.info("Setted up change alert");
+            updateStatus(ThingStatus.ONLINE);
+        } catch (NumberFormatException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR, "Pin not numeric");
+            logger.error("Non numeric pin number", e);
+        } catch (PigpioException e) {
+            if (e.getErrorCode() == PigpioException.PI_BAD_GPIO) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR, "Bad GPIO Pin");
+                logger.error("Bad GPIO Pin", e);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                        e.getLocalizedMessage());
+                logger.error("Unknown jpigpio exception", e);
```

Comment:
See also E.4 in the coding guidelines which logging levels are expected: http://docs.openhab.org/developers/development/guidelines.html#e-logging

Please check all `logger.error`'s